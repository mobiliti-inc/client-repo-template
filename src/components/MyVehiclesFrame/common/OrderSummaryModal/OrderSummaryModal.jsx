import moment from 'moment';


import DialogModal from '../../../../components/Modal/DialogModal/DialogModal';
import numberToUsCurrencyformatter from '../../../../utils/numberToUsCurrencyformatter';
import styles from './OrderSummaryModal.scss';

const OrderSummaryModal = (props) => (
	<DialogModal
		blueButtonClick={props.confirmSubscriptionMileChanges}
		whiteButtonClick={props.closeOrderSummaryModal}
		modalIsVisible={props.isOrderSummaryModalVisible}
		closeModal={props.closeOrderSummaryModal}
		modalTitle=""
		whiteButtonText="Cancel"
		blueButtonText="Confirm Change"
		isBlueButtoDisabled={false}
		footer
	>
		<div styleName="order-summary-container">
			<h2>Order Summary</h2>
			<p styleName="order-summary-info">
						Below are the changes for your upcoming subscription.
						You will be able to make edits to your subscription at
						anytime prior to your billing date.
			</p>
			<div styleName="order-summary-details">
				<span>New Charge Date </span>
				<span>{moment(props.charge_date, 'MM/DD/YYYY').format('ddd, MMM. Do')}</span>
			</div>
			<div styleName="order-summary-details">
				<span>Base Subscription Fee</span>
				<span>{`${numberToUsCurrencyformatter(2, 2).format(props.subscription_cost)}`}</span>
			</div>
			<div styleName="order-summary-details">
				<span>{`${props.monthly_miles} Miles Per Month`}</span>
				<span>{`+${numberToUsCurrencyformatter(2, 2).format(props.over_mileage_charge)}`}</span>
			</div>
			<div styleName="order-summary-details">
				<span>Taxes</span>
				<span>{`${numberToUsCurrencyformatter(2, 2).format(props.tax)}`}</span>
			</div>
			<div styleName="order-summary-details">
				<span>New Recurring Subscription Fee</span>
				<span>{`${numberToUsCurrencyformatter(2, 2).format(props.monthly_subscription_price)}`}</span>
			</div>
			<p styleName="exceed-miles-info">
				{`If you exceed your monthly miles you will be charged ${numberToUsCurrencyformatter(2, 2).format(props.excess_mile_charge)}
						per mile. Please refer to our FAQs for more information
						on how we charge for exceeding your monthly miles and exceeding
						your vehicle available miles`}
			</p>
		</div>
	</DialogModal>
);

OrderSummaryModal.propTypes = {
	closeOrderSummaryModal: PropTypes.func.isRequired,
	isOrderSummaryModalVisible: PropTypes.bool.isRequired,
	charge_date: PropTypes.string.isRequired,
	subscription_cost: PropTypes.number.isRequired,
	monthly_miles: PropTypes.number.isRequired,
	over_mileage_charge: PropTypes.number.isRequired,
	tax: PropTypes.number.isRequired,
	monthly_subscription_price: PropTypes.number.isRequired,
	excess_mile_charge: PropTypes.number.isRequired,
	confirmSubscriptionMileChanges: PropTypes.func.isRequired,
};

export default CSSModules(OrderSummaryModal, styles, { allowMultiple: true });
