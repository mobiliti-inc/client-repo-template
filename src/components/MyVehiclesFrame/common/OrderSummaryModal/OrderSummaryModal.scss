@import '~variables';

.order-summary-container {
    display: block;
    max-width: 500px;
	h2 {
		color: $darker-blue;
		margin-top: 0;
		font-size: 18px;
		font-weight: bold;
	}
	.order-summary-details {
		display: flex;
		justify-content: space-between;
		border-bottom: 1px solid rgba(0, 0, 0, 0.1);
		padding: 15px 0;
		span {
			font-family: $hanken-sans;
			font-size: 16px;
			color: $darker-blue;
		}
		span:last-child {
			font-weight: 600;
		}

		.total {
			color: $blue;
			font-weight: 600;
		}
	}

	.exceed-miles-info, .order-summary-info {
		font-family: $hanken-sans;
		color: $darker-blue;
		padding-top: 5px;
		font-size: 12px;
		font-weight: 300;
		margin-bottom: 0;
	}

	div:nth-last-child(2) {
		border-bottom: none;
	}
}
