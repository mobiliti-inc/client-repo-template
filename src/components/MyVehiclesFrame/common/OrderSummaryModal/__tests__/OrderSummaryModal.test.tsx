import OrderSummaryModal from '../OrderSummaryModal';

describe('OrderSummaryModal', () => {
	const props = {
		confirmSubscriptionMileChanges: jest.fn(),
		isFetching: false,
		subscription_id: '0',
		monthly_miles: 0,
		subscription_cost: 0,
		over_mileage_charge: 0,
		excess_mile_charge: 0,
		tax: 0,
		monthly_subscription_price: 0,
		charge_date: '00/00/000',
		closeOrderSummaryModal: jest.fn(),
		isOrderSummaryModalVisible: true,
	};
	const component = <OrderSummaryModal {...props} />;
	it('renders', () => {
		const subject = shallow(component);
		expect(subject).toMatchSnapshot();
	});
});
