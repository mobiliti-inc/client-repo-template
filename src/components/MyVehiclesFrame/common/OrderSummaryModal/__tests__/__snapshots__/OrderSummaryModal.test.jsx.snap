// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummaryModal renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderSummaryModal
    charge_date="00/00/000"
    closeOrderSummaryModal={[MockFunction]}
    confirmSubscriptionMileChanges={[MockFunction]}
    excess_mile_charge={0}
    isFetching={false}
    isOrderSummaryModalVisible={true}
    monthly_miles={0}
    monthly_subscription_price={0}
    over_mileage_charge={0}
    subscription_cost={0}
    subscription_id="0"
    tax={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "blueButtonClick": [MockFunction],
      "blueButtonText": "Confirm Change",
      "children": <div
        className="order-summary-container"
      >
        <h2>
          Order Summary
        </h2>
        <p
          className="order-summary-info"
        >
          Below are the changes for your upcoming subscription. You will be able to make edits to your subscription at anytime prior to your billing date.
        </p>
        <div
          className="order-summary-details"
        >
          <span>
            New Charge Date 
          </span>
          <span>
            Invalid date
          </span>
        </div>
        <div
          className="order-summary-details"
        >
          <span>
            Base Subscription Fee
          </span>
          <span>
            $0.00
          </span>
        </div>
        <div
          className="order-summary-details"
        >
          <span>
            0 Miles Per Month
          </span>
          <span>
            +$0.00
          </span>
        </div>
        <div
          className="order-summary-details"
        >
          <span>
            Taxes
          </span>
          <span>
            $0.00
          </span>
        </div>
        <div
          className="order-summary-details"
        >
          <span>
            New Recurring Subscription Fee
          </span>
          <span>
            $0.00
          </span>
        </div>
        <p
          className="exceed-miles-info"
        >
          If you exceed your monthly miles you will be charged $0.00
						per mile. Please refer to our FAQs for more information
						on how we charge for exceeding your monthly miles and exceeding
						your vehicle available miles
        </p>
      </div>,
      "closeModal": [MockFunction],
      "footer": true,
      "isBlueButtoDisabled": false,
      "modalIsVisible": true,
      "modalTitle": "",
      "whiteButtonClick": [MockFunction],
      "whiteButtonText": "Cancel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Order Summary
          </h2>,
          <p
            className="order-summary-info"
          >
            Below are the changes for your upcoming subscription. You will be able to make edits to your subscription at anytime prior to your billing date.
          </p>,
          <div
            className="order-summary-details"
          >
            <span>
              New Charge Date 
            </span>
            <span>
              Invalid date
            </span>
          </div>,
          <div
            className="order-summary-details"
          >
            <span>
              Base Subscription Fee
            </span>
            <span>
              $0.00
            </span>
          </div>,
          <div
            className="order-summary-details"
          >
            <span>
              0 Miles Per Month
            </span>
            <span>
              +$0.00
            </span>
          </div>,
          <div
            className="order-summary-details"
          >
            <span>
              Taxes
            </span>
            <span>
              $0.00
            </span>
          </div>,
          <div
            className="order-summary-details"
          >
            <span>
              New Recurring Subscription Fee
            </span>
            <span>
              $0.00
            </span>
          </div>,
          <p
            className="exceed-miles-info"
          >
            If you exceed your monthly miles you will be charged $0.00
						per mile. Please refer to our FAQs for more information
						on how we charge for exceeding your monthly miles and exceeding
						your vehicle available miles
          </p>,
        ],
        "className": "order-summary-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Order Summary",
          },
          "ref": null,
          "rendered": "Order Summary",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Below are the changes for your upcoming subscription. You will be able to make edits to your subscription at anytime prior to your billing date.",
            "className": "order-summary-info",
          },
          "ref": null,
          "rendered": "Below are the changes for your upcoming subscription. You will be able to make edits to your subscription at anytime prior to your billing date.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                New Charge Date 
              </span>,
              <span>
                Invalid date
              </span>,
            ],
            "className": "order-summary-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New Charge Date ",
              },
              "ref": null,
              "rendered": "New Charge Date ",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Invalid date",
              },
              "ref": null,
              "rendered": "Invalid date",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Base Subscription Fee
              </span>,
              <span>
                $0.00
              </span>,
            ],
            "className": "order-summary-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Base Subscription Fee",
              },
              "ref": null,
              "rendered": "Base Subscription Fee",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$0.00",
              },
              "ref": null,
              "rendered": "$0.00",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                0 Miles Per Month
              </span>,
              <span>
                +$0.00
              </span>,
            ],
            "className": "order-summary-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "0 Miles Per Month",
              },
              "ref": null,
              "rendered": "0 Miles Per Month",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "+$0.00",
              },
              "ref": null,
              "rendered": "+$0.00",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Taxes
              </span>,
              <span>
                $0.00
              </span>,
            ],
            "className": "order-summary-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Taxes",
              },
              "ref": null,
              "rendered": "Taxes",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$0.00",
              },
              "ref": null,
              "rendered": "$0.00",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                New Recurring Subscription Fee
              </span>,
              <span>
                $0.00
              </span>,
            ],
            "className": "order-summary-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New Recurring Subscription Fee",
              },
              "ref": null,
              "rendered": "New Recurring Subscription Fee",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$0.00",
              },
              "ref": null,
              "rendered": "$0.00",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "If you exceed your monthly miles you will be charged $0.00
						per mile. Please refer to our FAQs for more information
						on how we charge for exceeding your monthly miles and exceeding
						your vehicle available miles",
            "className": "exceed-miles-info",
          },
          "ref": null,
          "rendered": "If you exceed your monthly miles you will be charged $0.00
						per mile. Please refer to our FAQs for more information
						on how we charge for exceeding your monthly miles and exceeding
						your vehicle available miles",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "blueButtonClick": [MockFunction],
        "blueButtonText": "Confirm Change",
        "children": <div
          className="order-summary-container"
        >
          <h2>
            Order Summary
          </h2>
          <p
            className="order-summary-info"
          >
            Below are the changes for your upcoming subscription. You will be able to make edits to your subscription at anytime prior to your billing date.
          </p>
          <div
            className="order-summary-details"
          >
            <span>
              New Charge Date 
            </span>
            <span>
              Invalid date
            </span>
          </div>
          <div
            className="order-summary-details"
          >
            <span>
              Base Subscription Fee
            </span>
            <span>
              $0.00
            </span>
          </div>
          <div
            className="order-summary-details"
          >
            <span>
              0 Miles Per Month
            </span>
            <span>
              +$0.00
            </span>
          </div>
          <div
            className="order-summary-details"
          >
            <span>
              Taxes
            </span>
            <span>
              $0.00
            </span>
          </div>
          <div
            className="order-summary-details"
          >
            <span>
              New Recurring Subscription Fee
            </span>
            <span>
              $0.00
            </span>
          </div>
          <p
            className="exceed-miles-info"
          >
            If you exceed your monthly miles you will be charged $0.00
						per mile. Please refer to our FAQs for more information
						on how we charge for exceeding your monthly miles and exceeding
						your vehicle available miles
          </p>
        </div>,
        "closeModal": [MockFunction],
        "footer": true,
        "isBlueButtoDisabled": false,
        "modalIsVisible": true,
        "modalTitle": "",
        "whiteButtonClick": [MockFunction],
        "whiteButtonText": "Cancel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Order Summary
            </h2>,
            <p
              className="order-summary-info"
            >
              Below are the changes for your upcoming subscription. You will be able to make edits to your subscription at anytime prior to your billing date.
            </p>,
            <div
              className="order-summary-details"
            >
              <span>
                New Charge Date 
              </span>
              <span>
                Invalid date
              </span>
            </div>,
            <div
              className="order-summary-details"
            >
              <span>
                Base Subscription Fee
              </span>
              <span>
                $0.00
              </span>
            </div>,
            <div
              className="order-summary-details"
            >
              <span>
                0 Miles Per Month
              </span>
              <span>
                +$0.00
              </span>
            </div>,
            <div
              className="order-summary-details"
            >
              <span>
                Taxes
              </span>
              <span>
                $0.00
              </span>
            </div>,
            <div
              className="order-summary-details"
            >
              <span>
                New Recurring Subscription Fee
              </span>
              <span>
                $0.00
              </span>
            </div>,
            <p
              className="exceed-miles-info"
            >
              If you exceed your monthly miles you will be charged $0.00
						per mile. Please refer to our FAQs for more information
						on how we charge for exceeding your monthly miles and exceeding
						your vehicle available miles
            </p>,
          ],
          "className": "order-summary-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Order Summary",
            },
            "ref": null,
            "rendered": "Order Summary",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Below are the changes for your upcoming subscription. You will be able to make edits to your subscription at anytime prior to your billing date.",
              "className": "order-summary-info",
            },
            "ref": null,
            "rendered": "Below are the changes for your upcoming subscription. You will be able to make edits to your subscription at anytime prior to your billing date.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  New Charge Date 
                </span>,
                <span>
                  Invalid date
                </span>,
              ],
              "className": "order-summary-details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New Charge Date ",
                },
                "ref": null,
                "rendered": "New Charge Date ",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Invalid date",
                },
                "ref": null,
                "rendered": "Invalid date",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Base Subscription Fee
                </span>,
                <span>
                  $0.00
                </span>,
              ],
              "className": "order-summary-details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Base Subscription Fee",
                },
                "ref": null,
                "rendered": "Base Subscription Fee",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$0.00",
                },
                "ref": null,
                "rendered": "$0.00",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  0 Miles Per Month
                </span>,
                <span>
                  +$0.00
                </span>,
              ],
              "className": "order-summary-details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "0 Miles Per Month",
                },
                "ref": null,
                "rendered": "0 Miles Per Month",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "+$0.00",
                },
                "ref": null,
                "rendered": "+$0.00",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Taxes
                </span>,
                <span>
                  $0.00
                </span>,
              ],
              "className": "order-summary-details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Taxes",
                },
                "ref": null,
                "rendered": "Taxes",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$0.00",
                },
                "ref": null,
                "rendered": "$0.00",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  New Recurring Subscription Fee
                </span>,
                <span>
                  $0.00
                </span>,
              ],
              "className": "order-summary-details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New Recurring Subscription Fee",
                },
                "ref": null,
                "rendered": "New Recurring Subscription Fee",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$0.00",
                },
                "ref": null,
                "rendered": "$0.00",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "If you exceed your monthly miles you will be charged $0.00
						per mile. Please refer to our FAQs for more information
						on how we charge for exceeding your monthly miles and exceeding
						your vehicle available miles",
              "className": "exceed-miles-info",
            },
            "ref": null,
            "rendered": "If you exceed your monthly miles you will be charged $0.00
						per mile. Please refer to our FAQs for more information
						on how we charge for exceeding your monthly miles and exceeding
						your vehicle available miles",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
